### ISSUE TITLE
Error trying to read/open db: block too small for length field (rust integration)

### ISSUE THREAD

**Issue by jmingov** [peermaps/eyros#22]
Hi!

Great work with the project :)

I have bee using it for a personal project and I'm facing the following error trying to open the DB after about 1gb of data inserted:

Compat { error: ErrorMessage { msg: "block too small for length field" }

About the usage, I don't have any special calls to close the DB. I just reopen the same DB as the app gets launched with the  `DB::open_from_path` method.

The P, V in the db inserts are in the following format:

```
type P = (f64, f64, u8, i64);
type V = (u8, u8);
Row::Insert(point, value);
```

The data is just test data, not important, but i would like to know how/if I'm doing something wrong or how can it be prevented.

Br,
J

**Comment by substack** [comment]
I'm glad you're getting some use out of the db! That data looks fine, I think you are running into some long-standing bugs with the old version. I just published version 4.0.0 which I have stress tested much more (100+ million records), so the new version might fix your issue although the data format changed between versions.

**Comment by jmingov** [comment]
Hi there :) 

Thanks for the response.

No worries about the data format I can adapt to any change, its a little side project ^^.
My data size is now ~111M rows, Ill try with 4.0.0 and let you know of any issues! 

Best,
J
